src/renderers/dom.js

```diff
@@ -118,6 +118,13 @@ function addProps(el: HTMLElement | Element, node) {
   if (el.tagName.toLowerCase() === ELEMENT_TAG.IFRAME && !props.id) {
     el.setAttribute(ELEMENT_PROP.ID, `jsx-iframe-${uniqueID()}`);
   }
+
+  // If the element is an iframe and it has no srcdoc or src, set the srcdoc to an empty string.
+  // Content specified via srcdoc is treated as being from the same origin as the parent document.
+  // In some browsers (like Safari 17+), empty iframes without src or srcdoc may be treated as "anonymous," potentially blocking certain content.
+  if (el.tagName.toLowerCase() === ELEMENT_TAG.IFRAME && (!props.srcdoc && !props.src)) {
```

Comment:
Just a nit pick
```suggestion
  if (
      el.tagName.toLowerCase() === ELEMENT_TAG.IFRAME &&
      !props.srcdoc &&
      !props.src
    ) {
```

Reply:
Also in the longer term it might be nice to break out a `TAG_TO_DEFAULT_PROPS` mapping rather than inserting more conditions here.